src/main/scala/com/databricks/sparkdl/DeepImageFeaturizer.scala

```diff
@@ -0,0 +1,198 @@
+/*
+ * Copyright 2017 Databricks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.databricks.sparkdl
+
+import java.nio.file.{Files, Paths}
+
+import org.apache.spark.image.ImageSchema
+import org.apache.spark.ml.Transformer
+import org.apache.spark.ml.linalg.SQLDataTypes.VectorType
+import org.apache.spark.ml.linalg.Vectors
+import org.apache.spark.ml.param.{Param, ParamMap}
+import org.apache.spark.ml.util.{DefaultParamsWritable, Identifiable}
+import org.apache.spark.sql.{DataFrame, Dataset, Row}
+import org.apache.spark.sql.functions.{col, udf}
+import org.apache.spark.sql.types.StructType
+
+import org.tensorflow.framework.GraphDef
+import org.tensorframes.{Shape, ShapeDescription}
+import org.tensorframes.impl.{DebugRowOps, SerializedGraph, TensorFlowOps}
+
+
+class DeepImageFeaturizer(override val uid: String) extends Transformer with DefaultParamsWritable {
+
+  def this() = this(Identifiable.randomUID("deepImageFeaturizer"))
+
+  final val inputCol: Param[String] = new Param[String](this, "inputCol", "input column name")
+  final val outputCol: Param[String] = new Param[String](this, "outputCol", "output column name")
+
+  // TODO: make modelName a param
+  val modelName = "_test"
+  def getModelName: String = modelName
+
+  private val RESIZED_IMAGE_COL = "__sparkdl_imageResized"
+  private val INPUT_BUFFER_COL = "__sparkdl_imageBuffer"
+
+  private def validateSchema(schema: StructType) {
```

Comment:
just to confirm, this runs on the Dataset sent into transform right

Reply:
Yep. It's also used in `transformSchema` which, my understanding is, is used by Pipelines to ensure that the pipeline is valid (maybe during a call to pipeline.transform?).

Reply:
Yes, this is because the pipeline API tries to be more lazy than what Spark allows, especially when combining estimators into a pipeline.

Nit by the way: `: Unit = {` As written, it will not be compatible with Scala 3.

Reply:
Scala 3 requires all methods to have return types?